### PR TITLE
PUBDEV-8585: verify GLM binomial IRLSM implementation and p-value calculation

### PR DESCRIPTION
**Pull request by wendycwong**
This PR completes the task in JIRA: https://h2oai.atlassian.net/browse/PUBDEV-8585

There are two goals here:
1. Make sure our p-value calculation matches the one in  https://stats.stackexchange.com/questions/89484/how-to-compute-the-standard-errors-of-a-logistic-regressions-coefficients ;
2. Make sure our binomial IRLSM implementation matches the one described in the elements of statistical learning.

Regarding 1:
I checked the covariance matrix calculation found that we have implemented the same calculation in our p-value calculation.  To prove that I added a java test to manually calculate the covariance matrix, the standard error, the z-value and then the p-value.  I then compared this calculation with the on in the GLM code and they match well.  I even use a different method to perform the equivalent of finding the inverse of the covariance matrix.
In addition, I verified the correct implementation of the p-value calculation when standardize=false as well.

Regarding 2:
I have checked and verified correct binomial IRLSM implementation matching the one in ESL for both when standardize=false and standardize=true

### CODE REVIEW

**Comment by wendycwong** [comment]
Benchmark results:
image

**Comment by wendycwong** [comment]
Really should be no difference in result because compute_p_values is not enabled.

**Review by ledell — CHANGES_REQUESTED** [review]
Hi @wendycwong just some small feedback on the param descriptions.  Can you specify explicitly what a default of `-1` means?  (e.g. disabled, or something else)

File: h2o-py/h2o/estimators/model_selection.py [hunk 1]
```diff
@@ -64,8 +64,8 @@ def __init__(self,
                  intercept=False,  # type: bool
                  non_negative=False,  # type: bool
                  max_iterations=0,  # type: int
-                 objective_epsilon=0.0,  # type: float
-                 beta_epsilon=0.0,  # type: float
+                 objective_epsilon=-1.0,  # type: float
```
Comment (ledell) [inline h2o-py/h2o/estimators/model_selection.py:67]: These should be -1 not -1.0.

**Comment by ledell** [comment]
@wendycwong My requested changes are minor (not blockers) so if you want this to go into 3.36.1.1 w/o the changes, that's ok with me (can easily follow-up later).

**Review by ledell — APPROVED** [review]
Approving so you can merge if you need to on Monday (see my notes above about improving param descriptions and -1.0 default value on Python client.

File: h2o-py/h2o/estimators/model_selection.py (hunk 1 above)
Comment (wendycwong) [reply h2o-py/h2o/estimators/model_selection.py:67]: yes, they are of type double.

**Pull request merged**